File: ScalaStudio/01-basics/basics.scala
object Basics {
  def main(args: Array[String]): Unit = {
    println("=== 1. 变量与常量 ===")
    val name: String = "ScalaStudio" // 不可变
    var version: Double = 3.3      // 可变
    version = 3.4
    
    println(s"项目名称: $name")
    println(s"版本号: $version")

    println("\n=== 2. 基本数据类型 ===")
    val age: Int = 25
    val pi: Double = 3.14159
    val flag: Boolean = true
    println(s"Int: $age, Double: $pi, Boolean: $flag")
  }
}
Basics.main(Array())

File: HaskellStudio/01-basics/Basics.hs
module Basics where

runBasics :: IO ()
runBasics = do
    putStrLn "=== 1. Haskell 基础与变量 ==="
    let name = "HaskellStudio"
    let version = 9.6
    putStrLn ("语言名称: " ++ name)
    putStrLn ("版本: " ++ show version)
    
    putStrLn "\n=== 2. 基本运算 ==="
    let sumVal = 10 + 20
    let prodVal = 5 * 6
    putStrLn ("10 + 20 = " ++ show sumVal)
    putStrLn ("5 * 6 = " ++ show prodVal)
